crates/ruff/src/registry.rs

```diff
@@ -1,672 +1,16 @@
-    rules::flake8_use_pathlib::violations::OsPathJoin,
-    rules::flake8_use_pathlib::violations::OsPathBasename,
-    rules::flake8_use_pathlib::violations::OsPathDirname,
-    rules::flake8_use_pathlib::violations::OsPathSamefile,
-    rules::flake8_use_pathlib::violations::OsPathSplitext,
-    rules::flake8_use_pathlib::violations::BuiltinOpen,
-    rules::flake8_use_pathlib::violations::PyPath,
-    // flake8-logging-format
-    rules::flake8_logging_format::violations::LoggingStringFormat,
-    rules::flake8_logging_format::violations::LoggingPercentFormat,
-    rules::flake8_logging_format::violations::LoggingStringConcat,
-    rules::flake8_logging_format::violations::LoggingFString,
-    rules::flake8_logging_format::violations::LoggingWarn,
-    rules::flake8_logging_format::violations::LoggingExtraAttrClash,
-    rules::flake8_logging_format::violations::LoggingExcInfo,
-    rules::flake8_logging_format::violations::LoggingRedundantExcInfo,
-    // flake8-raise
-    rules::flake8_raise::rules::UnnecessaryParenOnRaiseException,
-    // flake8-self
-    rules::flake8_self::rules::PrivateMemberAccess,
-    // flake8-gettext
-    rules::flake8_gettext::rules::FStringInGetTextFuncCall,
-    rules::flake8_gettext::rules::FormatInGetTextFuncCall,
-    rules::flake8_gettext::rules::PrintfInGetTextFuncCall,
-    // numpy
-    rules::numpy::rules::NumpyDeprecatedTypeAlias,
-    rules::numpy::rules::NumpyLegacyRandom,
-    // ruff
-    rules::ruff::rules::AmbiguousUnicodeCharacterString,
-    rules::ruff::rules::AmbiguousUnicodeCharacterDocstring,
-    rules::ruff::rules::AmbiguousUnicodeCharacterComment,
-    rules::ruff::rules::CollectionLiteralConcatenation,
-    rules::ruff::rules::AsyncioDanglingTask,
-    rules::ruff::rules::UnusedNOQA,
-    rules::ruff::rules::PairwiseOverZipped,
-    rules::ruff::rules::MutableDataclassDefault,
-    rules::ruff::rules::FunctionCallInDataclassDefaultArgument,
-    rules::ruff::rules::ExplicitFStringTypeConversion,
-    rules::ruff::rules::InvalidPyprojectToml,
-    // flake8-django
-    rules::flake8_django::rules::DjangoNullableModelStringField,
-    rules::flake8_django::rules::DjangoLocalsInRenderFunction,
-    rules::flake8_django::rules::DjangoExcludeWithModelForm,
-    rules::flake8_django::rules::DjangoAllWithModelForm,
-    rules::flake8_django::rules::DjangoModelWithoutDunderStr,
-    rules::flake8_django::rules::DjangoUnorderedBodyContentInModel,
-    rules::flake8_django::rules::DjangoNonLeadingReceiverDecorator,
-    // flynt
-    rules::flynt::rules::StaticJoinToFString,
-    // flake8-todo
-    rules::flake8_todos::rules::InvalidTodoTag,
-    rules::flake8_todos::rules::MissingTodoAuthor,
-    rules::flake8_todos::rules::MissingTodoLink,
-    rules::flake8_todos::rules::MissingTodoColon,
-    rules::flake8_todos::rules::MissingTodoDescription,
-    rules::flake8_todos::rules::InvalidTodoCapitalization,
-    rules::flake8_todos::rules::MissingSpaceAfterTodoColon,
-);
-
 pub trait AsRule {
```

Comment:
Can / should this and `impl Rule` be in `codes.rs`?

Reply:
yes, but i'd do that in a separate PR